doc/contribute/index.rst

```diff
@@ -331,6 +331,37 @@ following exceptions:
   comment, ``//``, is not allowed.
 * Use ``/**  */`` for doxygen comments that need to appear in the documentation.
 
+Identifier Naming
+=================
+
+* Names of macros defining constants and labels in enums are all capital
+  letters and underscores.
+* Functions,  and function-like macros use all lowercase, with underscores, of
```

Comment:
So this is different from what we have been doing to date in Zephyr.
Currently the form is module_noun_[noun...]_verb, with nouns in increasing order of specificity and the verb last.
Unless you want to go through and rename a bunch of APIs in the kernel (and deal with deprecation headaches for the public ones) I think we shouldn't change this.